sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -478,15 +478,15 @@ final class Decimal extends Ordered[Decimal] with Serializable {
     if (decimalVal.eq(null) && that.decimalVal.eq(null) && scale == that.scale) {
       Decimal(longVal + that.longVal, Math.max(precision, that.precision), scale)
     } else {
-      Decimal(toBigDecimal + that.toBigDecimal)
```

Comment:
In Scala 2.13, `+` method is 
```
def +  (that: BigDecimal): BigDecimal = new BigDecimal(this.bigDecimal.add(that.bigDecimal, mc), mc)
```

and in Scala 2.12 `+` method is 
```
def +  (that: BigDecimal): BigDecimal = new BigDecimal(this.bigDecimal add that.bigDecimal, mc)
```

There are some differences in accuracy.

Reply:
I don't think we want to set a MathContext here anyway?

Reply:
Do you mean we need change to use methods with MathContext ? Like `BigDecimal add(BigDecimal augend, MathContext mc) `?

Reply:
Sorry , I think I don't fully understand this comments ....

Reply:
I think the change is OK here, because we actually do not want to modify the rounding, right?

Reply:
Yes, you are right ~